frontend/src/metabase-lib/Question.ts

```diff
@@ -194,7 +189,13 @@ class Question {
       console.warn("Unknown query type: " + datasetQuery?.type);
   });
 
-  legacyQuery({ useStructuredQuery }: { useStructuredQuery?: boolean } = {}) {
+  legacyQuery<UseStructuredQuery extends boolean>({
```

Comment:
This is generic to help with the following error in `filter` call in `ObjectDetailView`:

```
Property 'filter' does not exist on type 'AtomicQuery | StructuredQuery'.
```